tests/test_raw_node.rs

```diff
@@ -141,6 +141,56 @@ fn test_raw_node_propose_and_conf_change() {
     assert_eq!(entries[1].get_data(), &*ccdata);
 }
 
+// test_raw_node_read_index ensures that RawNode.read_index sends the MsgReadIndex message
+// to the underlying raft. It also ensures that ReadState can be read out.
+#[test]
+fn test_raw_node_read_index() {
```

Comment:
In either way, the test case is different with etcd/raft, because it doesn't has a `allow_step`. I'm not sure whether it's a good idea to add a function call overhead to raft for every message, just for making a simple test case look like the same to etcd/raft. Even it's doable.